pkg/lib/authn/authenticator/oob/provider.go

```diff
@@ -61,34 +74,44 @@ func (p *Provider) Create(a *Authenticator) error {
 	return p.Store.Create(a)
 }
 
-func (p *Provider) getOTPOpts(channel authn.AuthenticatorOOBChannel) otp.ValidateOpts {
-	var digits int
-	switch channel {
-	case authn.AuthenticatorOOBChannelEmail:
-		digits = p.Config.Email.CodeDigits
-	case authn.AuthenticatorOOBChannelSMS:
-		digits = p.Config.SMS.CodeDigits
-	default:
-		panic("oob: unknown channel type: " + channel)
-	}
-	return otp.ValidateOptsOOBTOTP(digits)
+func (p *Provider) GetCode(authenticatorID string) (*Code, error) {
+	return p.CodeStore.Get(authenticatorID)
 }
 
-func (p *Provider) Authenticate(secret string, channel authn.AuthenticatorOOBChannel, code string) error {
-	ok := otp.ValidateTOTP(secret, code, p.Clock.NowUTC(), p.getOTPOpts(channel))
-	if !ok {
-		return errors.New("invalid OOB code")
+func (p *Provider) CreateCode(authenticatorID string) (*Code, error) {
+	code := otp.FormatNumeric.Generate()
+	codeModel := &Code{
+		AuthenticatorID: authenticatorID,
+		Code:            code,
+		// TODO(oob): Expiry should be configurable
+		ExpireAt: p.Clock.NowUTC().Add(time.Duration(3600) * time.Second),
```

Comment:
Do you think we need config for the oob code expiry? Or too avoid too many config from now, so set a fixed value is fine?